public/js/Histogram.js

```diff
@@ -0,0 +1,100 @@
+var d3 = require('d3');
+var _ = require('lodash');
+_.mixin(require('lodash-deep'));
+
+function Histogram(sampleData, element, featureNames) {
```

Comment:
Prototypical inheritance just means that objects inherit from other objects. This contrasts to classical inheritance where classes inherit from other classes, and these classes in turn define objects. Prototypical inheritance has some advantages -- it's simple yet powerful. It's possible to easily extend objects without a whole bunch of extra infrastructure code. This is useful in a dynamic environment like JavaScript. Everything just inherits from an ancestor object.

Say I want to extend the Array object to add this awesome new method:

```
> Array.prototype.sayHello = function() {
>    console.log('I\'m just an array, please don\'t slice and splice me!');
> }
> [1, 2, 3, 4].sayHello()
I'm just an array, please don't slice and splice me!
```

In the classical inheritance model, I'd need to define a new class that extends the Array class to do that, THEN create a new instance of that new extended class.

Technically, JavaScript actually implements a *pseudo-classical* inheritance model. That's why there's the constructor function. It's not really 'constructing' anything in the traditional sense where a class definition is used to build a new object. It's just creating an object, and putting all those properties you've defined into it. Then when the function is called using the ``new`` keyword, a new instance of that object is created, ie the new object inherits from an ancestor object. 

These "constructors" are differentiated from normal functions by capitalising the first letter of their name. Ditto how classes are named in other OOP languages. This isn't enforced, but it's pretty much universally accepted as the way to do it and will be enforced by default in all JS linters.

ES6, the new version of JavaScript is going to introduce a ``class`` keyword. It's just syntactic sugar and doesn't change the fundamental nature of JS but I'm not big on it. I think it'll just confuse matters more, especially if you're a programmer coming from a classical inheritance background.